sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CaseDirectorySteps.java

```diff
@@ -127,14 +141,269 @@ public CaseDirectorySteps(
+        () -> {
+          String differentPresentCondition = apiState.getLastCreatedPerson().getPresentCondition();
+          while (differentPresentCondition.equals(
+              apiState.getLastCreatedPerson().getPresentCondition())) {
+            differentPresentCondition = PresentCondition.getRandomPresentCondition();
+          }
           webDriverHelpers.selectFromCombobox(
-              CASE_OUTCOME_FILTER_COMBOBOX, CaseOutcome.getValueFor(outcomeFilterOption));
-          webDriverHelpers.clickOnWebElementBySelector(CASE_APPLY_FILTERS_BUTTON);
+              CASE_PRESENT_CONDITION_COMBOBOX,
+              PresentCondition.getValueFor(differentPresentCondition));
+        });
+    And(
+        "I click APPLY BUTTON in Case Directory Page",
+        () -> {
+          webDriverHelpers.clickOnWebElementBySelector(
+              CASE_DIRECTORY_DETAILED_PAGE_APPLY_FILTER_BUTTON);
+          TimeUnit.SECONDS.sleep(3); // needed for table refresh
+        });
+
+    Then(
+        "I apply Case origin {string}",
+        (String caseOrigin) ->
+            webDriverHelpers.selectFromCombobox(
+                CASE_ORIGIN_FILTER_COMBOBOX, CaseOrigin.getValueFor(caseOrigin)));
+    Then(
+        "I apply Community {string}",
+        (String community) ->
+            webDriverHelpers.selectFromCombobox(CASE_COMMUNITY_FILTER_COMBOBOX, community));
+
+    And(
+        "I apply Region filter {string}",
+        (String region) ->
+            webDriverHelpers.selectFromCombobox(CASE_REGION_FILTER_COMBOBOX, region));
+    And(
+        "I apply Surveillance Officer filter {string}",
+        (String survoff) ->
+            webDriverHelpers.selectFromCombobox(CASE_SURVOFF_FILTER_COMBOBOX, survoff));
+    And(
+        "I apply Reporting User filter {string}",
+        (String reportingUser) ->
+            webDriverHelpers.fillInWebElement(CASE_REPORTING_USER_FILTER, reportingUser));
+    And(
+        "I apply Vaccination Status filter to {string}",
+        (String vaccinationStatus) ->
+            webDriverHelpers.selectFromCombobox(
+                CASE_VACCINATION_STATUS_FILTER_COMBOBOX, vaccinationStatus));
+    And(
+        "I apply Quarantine filter to {string}",
+        (String quarantine) ->
+            webDriverHelpers.selectFromCombobox(CASE_QUARANTINE_FILTER_COMBOBOX, quarantine));
+    And(
+        "I apply Reinfection filter to {string}",
+        (String reinfection) ->
+            webDriverHelpers.selectFromCombobox(CASE_REINFECTION_FILTER_COMBOBOX, reinfection));
+    And(
+        "I apply Date type filter to {string}",
+        (String dataType) ->
+            webDriverHelpers.selectFromCombobox(CASE_DATA_TYPE_FILTER_COMBOBOX, dataType));
+
+    And(
+        "I fill Cases from input to day before mocked case created",
+        () -> {
+          DateTimeFormatter formatter = DateTimeFormatter.ofPattern("M/d/yyyy");
+          webDriverHelpers.fillInWebElement(
```

Comment:
based on the action, I assume that it's always checking for today's date, and does the minus 3, but if we want to reuse it in other scenarios/tests, it won't apply. So, can we take the date from entity and do the -3? Same, the name of the entity should be present in step name